server/models/Update.ts

```diff
@@ -0,0 +1,538 @@
+  public declare html: string;
+  public declare summary: string;
+  public declare image: string;
+  public declare isPrivate: boolean;
+  public declare isChangelog: boolean;
+  public declare notificationAudience: (typeof UPDATE_NOTIFICATION_AUDIENCE)[keyof typeof UPDATE_NOTIFICATION_AUDIENCE];
+  public declare tags: string[];
+
+  public declare publishedAt: Date;
+  public declare makePublicOn: Date;
+  public declare createdAt: Date;
+  public declare updatedAt: Date;
+  public declare deletedAt: Date;
+
+  public declare collective?: typeof models.Collective;
+  public declare fromCollective?: typeof models.Collective;
+
+  public declare getCollective: BelongsToGetAssociationMixin<Collective>;
+  public declare getFromCollective: BelongsToGetAssociationMixin<Collective>;
+  public declare getCreatedByUser: BelongsToGetAssociationMixin<User>;
+  public declare getLastEditedByUser: BelongsToGetAssociationMixin<User>;
+
+  /**
+   * Instance Methods
+   */
+
+  // Edit an update
+  edit = async function (remoteUser, newUpdateData) {
+    if (newUpdateData.TierId) {
+      const tier = await models.Tier.findByPk(newUpdateData.TierId);
+      if (!tier) {
+        throw new errors.ValidationFailed('Tier not found');
+      }
+      if (tier.CollectiveId !== this.CollectiveId) {
+        throw new errors.ValidationFailed("Cannot link this update to a Tier that doesn't belong to this collective");
+      }
+    }
+
+    const editableAttributes = ['TierId', 'title', 'html', 'tags', 'isPrivate', 'isChangelog', 'makePublicOn'];
+
+    return await this.update({
+      ...pick(newUpdateData, editableAttributes),
+      LastEditedByUserId: remoteUser.id,
+    });
+  };
+
+  // Publish update
+  publish = async function (remoteUser, notificationAudience) {
+    this.publishedAt = new Date();
+    this.notificationAudience = notificationAudience;
+    this.collective = this.collective || (await models.Collective.findByPk(this.CollectiveId));
+    this.fromCollective = this.fromCollective || (await models.Collective.findByPk(this.FromCollectiveId));
+
+    models.Activity.create({
+      type: activities.COLLECTIVE_UPDATE_PUBLISHED,
+      UserId: remoteUser.id,
+      CollectiveId: this.CollectiveId,
+      FromCollectiveId: this.FromCollectiveId,
+      HostCollectiveId: this.collective.approvedAt ? this.collective.HostCollectiveId : null,
+      data: {
+        fromCollective: this.fromCollective.activity,
+        collective: this.collective.activity,
+        update: this.activity,
+        url: `${config.host.website}/${this.collective.slug}/updates/${this.slug}`,
+      },
+    });
+    return await this.save();
+  };
+
+  // Unpublish update
+  unpublish = async function (remoteUser) {
+    return this.update({ LastEditedByUserId: remoteUser.id, publishedAt: null });
+  };
+
+  delete = async function (remoteUser) {
+    await models.Comment.destroy({ where: { UpdateId: this.id } });
+    await models.Update.update(
```

Comment:
Should this be `this.update` instead?

Reply:
No, it's a `Model.update(attributes, { where: ... })`. We could write it as:

```es6
update = await models.find(...);
await update.destroy();
```

But there wouldn't be much added value, and I don't want to dig into this kind of changes here, I want to keep the diff minimal to prevent introducing side-effects